rasa/core/nlg/callback.py

```diff
@@ -62,7 +62,7 @@ def nlg_request_format(
     tracker_state = tracker.current_state(EventVerbosity.ALL)
 
     return {
-        "template": template_name,
+        "response": utter_action,
```

Comment:
Updated. Maybe we should also change the test to assert that both are there. Is this an external endpoint? I don't seem to find it here: https://rasa.com/docs/rasa/pages/http-api

Reply:
>  Maybe we should also change the test to assert that both are there

When they add a NLG server, then we send a HTTP request to the NLG server which then hopefully processes the request and returns bot responses. So it's not a Rasa Open Source endpoint, but another endpoint which we call. This is similar as we call the Rasa SDK to give us a bunch of events in exchange for the name of a custom action.

Reply:
Is there doc / catalog for such endpoints?

Reply:
Notting apart from this [docs page](https://rasa.com/docs/rasa/nlg). What are you looking for / expecting?

Reply:
This doc covered it. I think I finally internalized what's happening. 😅 I was confused about which components can get swapped out and the direction of the requests. 

In this case, is it even safe to just support both? If the NLG server has some kind of schema validation, it would fail, no? Do we also add a warning?

Reply:
I was looking for the object schemas that specify how the fields should look like, which fields can be null / if the numbers can be negative etc. Like the API spec, but for hooks.

Reply:
>  So this is actually a hook right?

Yes, exactly 👍🏻 

> In this case, is it even safe to just support both? If the NLG server has some kind of schema validation, it would fail, no?

haha, good point! Extra fields are usually fine, no (thinking of our Rasa X schema validation)?

Reply:
Yeah, we didn't have a "contract" in the first place, I think it's good enough. 😁